corehq/apps/reports/static/reports/js/scheduled_reports_list.js

```diff
@@ -0,0 +1,185 @@
+/*
+
+    ScheduledReportModel - model representing a single row on the list / a single report filter file
+    ScheduleReportPanelModel - model representing a whole table ('My Scheduled Reports' and 'Other Scheduled Reports')
+    ScheduledReportListModel - model representing the whole page (list of reports, actions)
+
+*/
+
+hqDefine("reports/js/scheduled_reports_list", [
+    'jquery',
+    'knockout',
+    'underscore',
+    'hqwebapp/js/components.ko', // pagination & feedback widget
+
+], function (
+    $,
+    ko,
+    _,
+) {
+
+    var scheduledReportModel = function (data, isOwner, isAdmin) {
+
+        var self = ko.mapping.fromJS(data);
+
+        self.configMany = ko.computed(function() {
+            if (data.configs.length === 1){
+                return false;
+            };
+            return true;
+        });
+
+        self.recipient_email_count = ko.computed(function() {
+            return self.recipient_emails().length;
+        });
+
+        self.deleteScheduledReport = function(observable, event){
+            $(event.currentTarget).closest('form').submit();
+        };
+
+        self.is_owner = isOwner;
+        self.is_admin = isAdmin
+        self.firstConfig = ko.observable(data.configs[0]);
+
+        return self;
+    };
+
+    var scheduledReportsPanelModel = function (options) {
```

Comment:
for all of the `options` and initial values passed into the object, it's good practice to assert what is required/optional, just for future use / reading of the models.

see: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/hqwebapp/static/hqwebapp/js/assert_properties.js#L1

example: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/users/static/users/js/web_users.js#L25

Reply:
good call - added!